### ISSUE TITLE
Completion Engine throw error in some specific real cases 

### ISSUE THREAD

**Issue by VincentBlondeau** [guillep/pharo-core#5615]
I wanted to modify a method, which is, I agree, quite complex, and there the completion engine threw me an error. 

You can add a test in 

> CompletionContextTest

that tries to create a context for this code:

```
methodSummariesDo: aBlock meta: aBoolean summaryBuilder: aSummaryBuilder
	targetClass ifNi:
		ifTrue: [ ^ #() ].
	self methodDictionary
		keysAndValuesDo: [ :selector :method | 
			| exportMethod packageName |
			targetClass isClass
				ifTrue: [ self
						ifTrue: [ method origin = TAssertable class
								ifTrue: [ exportMethod := false ]
								ifFalse: [ exportMethod := true.
									packageName := #'' ] ]
						ifFalse: [ exportMethod := true.
							packageName := #'' ] ]
				ifFalse:
					[ packageName := self packageName.
					packageName notNil
						ifTrue: [ | extension |
							extension notNil
								ifTrue: [ 
									exportMethod
										ifFalse: [ 0
												xtranscript:
													''Ignore '' , self name , '' '' , selector ] ]
								ifFalse: [ exportMethod := true ] ]
						ifFalse: [ exportMethod := false ] ] ]
```

Both fails under Pharo 8 and 9

**Comment by Ducasse** [comment]
Thanks. We got regressions with the new completion so it is good to have bug reports.

**Comment by MarcusDenker** [comment]
The problem is that #commonSuperclassWith: was not implemented correctly... we need the semantics that if we call it on a class that is already the superclass of the argument, it should return that.

ProtoObject commonSuperclassWith: Object  -> ProtoObject

**Comment by MarcusDenker** [comment]
https://github.com/pharo-project/pharo/pull/5643

**Comment by MarcusDenker** [comment]
(I did not add the "fixes ..." as this is a good fix to back port)

**Comment by myroslavarm** [comment]
This is not closed yet... because it hasn't been implemented in Pharo 8 or ?

**Comment by MarcusDenker** [comment]
yes, could be... we should check  commonSuperclassWith:  in Pharo8

**Comment by guillep** [comment]
So this is fixed in #5643. The backport issue is in #6418 . We can close this one now.
